canals/component/sockets.py

```diff
@@ -1,23 +1,29 @@
 # SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
 #
 # SPDX-License-Identifier: Apache-2.0
-from typing import Optional, Union, get_origin, get_args
 import logging
+import re
+from typing import List, Union, get_origin, get_args
 from dataclasses import dataclass, field
 
 
 logger = logging.getLogger(__name__)
 
+# Any input name ending with 3 underscores (e.g. "foo_bar___") will be marked as variadic
+VARIADIC_REGEX = re.compile(r"^\w+_{3}$")
```

Comment:
The current implementation derives whether or not a socket is variadic depending on its name. As variadic args in C are expressed with three dots `...`, I thought we could "translate" it into three underscores `___`.

The alternative to detect the nature of a socket from its name is adding a `set_variadic_input_types` method to the `component` api. That was my first implementation but I stumbled upon a few issues:
1. currently `set_input_types` overwrites `__canals_input__` but we can expect inputs being some variadic and some not, this would call for either changing the `set_input_types` api in a breaking way, or adding a second `set_variadic_input_types` api.
2. it would be impossible to deduce the variadic nature of an input based only on the `run` signature

Reply:
> it would be impossible to deduce the variadic nature of an input based only on the run signature

We could do this sort of signaling with types actually. `def run(self, variadic_param: Variadic[str])`

Reply:
Done!